tests/pipelines/test_pipelines_auto.py

```diff
@@ -0,0 +1,94 @@
+# coding=utf-8
+# Copyright 2023 HuggingFace Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from collections import OrderedDict
+
+from diffusers import (
+    AutoPipelineForImage2Image,
+    AutoPipelineForInpainting,
+    AutoPipelineForText2Image,
+)
+from diffusers.pipelines.auto_pipeline import (
+    AUTO_IMAGE2IMAGE_PIPELINES_MAPPING,
+    AUTO_INPAINTING_PIPELINES_MAPPING,
+    AUTO_TEXT2IMAGE_PIPELINES_MAPPING,
+)
+from diffusers.utils import slow
+
+
+PRETRAINED_MODEL_REPO_MAPPING = OrderedDict(
+    [
+        ("stable-diffusion", "runwayml/stable-diffusion-v1-5"),
+        ("if", "DeepFloyd/IF-I-XL-v1.0"),
+        ("kandinsky", "kandinsky-community/kandinsky-2-1"),
+        ("kdnsinskyv22", "kandinsky-community/kandinsky-2-2-decoder"),
+    ]
+)
+
+class AutoPipelineFastTest(unittest.TestCase):
+
+    def test_from_pipe_consistent(self):
+        pipe = AutoPipelineForText2Image.from_pretrained("hf-internal-testing/tiny-stable-diffusion-pipe", requires_safety_checker=False)
+        original_config = dict(pipe.config)
+
+        pipe = AutoPipelineForImage2Image.from_pipe(pipe)
+        pipe = AutoPipelineForText2Image.from_pipe(pipe)
+
+        assert dict(pipe.config) == original_config
```

Comment:
@yiyixuxu this test currently fails, but it shouldn't. The reason is that we don't correctly forward `requires_safety_checker`. 

Note that there might even be harder examples in case we go from SD-XL Image-2-Image over Text-2-Image back to SD-XL Image-2-Image.

Could you think about how we can make this work? We should probably just make sure we correctly save all the optional kwargs in `from_pipe` with `register_to_config`

Reply:
@patrickvonplaten updated `from_pipe` here